bears/general/IndentationBear.py

```diff
@@ -153,6 +182,64 @@ def run(self,
                     affected_code=(diff.range(filename),),
                     diffs={filename: diff})
 
+    def _get_indent_keywords_range(self,
+                                   file,
+                                   filename,
+                                   indent_keywords,
+                                   ranges):
+        keyword_pos = {}
+        for keyword in indent_keywords:
+            keyword_pos[keyword] = (tuple(get_valid_sequences(file,
+                                                              keyword,
+                                                              ranges,
+                                                              keyword=True)))
+        keyword_range = []
+        for keyword in keyword_pos:
+            for position in keyword_pos[keyword]:
+                start = position
+                end = AbsolutePosition(file, position.position + len(keyword))
+                for encapsulator in ranges["encapsulators"]:
+                    if (encapsulator.start.line == position.line and
+                            encapsulator.end.line > end.line):
+                        end = encapsulator.end
+                indent = True
+                for _range in ranges["indent_ranges"]:
+                    if(_range.start.line == end.line and
+                            _range.start.column > position.column):
+                        indent = False
+                        break
+                if indent:
+                    keyword_range.append(SourceRange.from_values(
+                        filename,
+                        start_line=start.line,
+                        start_column=start.column,
+                        end_line=end.line,
+                        end_column=end.column))
+        return keyword_range
+
+    def _get_keyword_indented_file(self,
+                                   file,
+                                   keyword_ranges,
+                                   ranges,
+                                   insert):
+        for _range in keyword_ranges:
+            prev_indent = get_indent_of_line(file,
+                                             _range.start.line - 1,
+                                             length=False)
+            file[_range.end.line] = (prev_indent + insert +
+                                     file[_range.end.line].lstrip())
+            start = _range.end
+            end = _range.end
+            for encapsulator in ranges["encapsulators"]:
```

Comment:
isn't this the same (or very similar) to the code above? If you can move the finding of `end` into an own function, maybe you can use it in both places?